keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1229 @@
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+#### Compute Specification Option
+
+The CCI model combined with some of the capabilities introduced in the Dynamic 
+Resource Allocation (DRA) KEP [3], offers the ability to transition compute 
+resource allocation behavior from being a cluster-admin dominated configuration
+to one that allows users with precise compute requirements to articulate the
+compute attributes that they need. 
+
+In several domains such Streaming, Telco, HPC, and AI/ML, users require
+fine-grained compute resource control to maximize their key performance 
+indicators.  They need an API option that can be made available through
+upstream Kubernetes components and specific compute resource allocation 
+plugins.
+
+##### Example 1: Attributed-based resource specification
+
+One realization of such an API can be achieved by an attribute-based 
+mechanism for compute resources that resembles the Dynamic Resource 
+Allocation (DRA) claim mechanism. The attribute-based compute resource 
+request model is a step towards improved semantic workload portability 
+that can be tried in a progressive manner with CCI. 
+
+In the following, there is an example of a novel approach that could be 
+fulfilled with the CCI architecture. This example presents a set of
+per-core attributes that enable precision beyond a policy-only based approach: 
+
+    # A resource request which consists of:
+    # 2 exclusive cores (no other processes running on them)
+    # 	* the cores shall be siblings
+    # 	* they shall run at 2.1 ghz
+    # 	* application assigned has the highest priority on the core
+    # 	* there are no IRQs on the core
+    # 6 exclusive cores(no other processes running on them)
+    # 4 shared cores(other processes can run on them) 
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: example-compute-claim-parameters
+      namespace: default
+    data:
+      cores: |
+    2
+    6
+    4
+      compute_attributes: |
+        exclusive, smt-sibling-required, 2.1Ghz, 1.0 priority, no-irq
+        exclusive
+        shared
+    apiVersion: resource.k8s.io/v1alpha1
+    kind: ResourceClaimTemplate
+    metadata:
+      name: example-compute-claim-parameters-template
+      namespace: default
+    spec:
+      metadata:
+        labels:
+          app: inline-resource
+      spec:
+        resourceClassName: cpuressources
+        parametersRef:
+          kind: ConfigMap
+          name: example-compute-claim-parameters
```

Comment:
In this example, will the DRA driver referenced by the `cpuressources` ResourceClass actually influence the scheduling decision?

In other words, does it really "allocate" these resources somehow?

If not, then using a `ResourceClaimTemplate` just to reference some additional parameters will be overkill and slow down scheduling.

To make the example more complete and realistic, it would be good to show what the PodSpec looks like and use a CRD instead of a ConfigMap for the parameters.

Reply:
I can't imaging what a DRA driver would look like that does this, given that the resources mentioned in this example (CPUs) are managed by kubelet+kube-scheduler.

When I first looked at this, my impression was that this KEP would explain how to do this and would propose how to implement such a DRA driver.

Now it is clear that this isn't the purpose of the KEP, so IMHO this example is more misleading than helpful. Can you take it out?

Reply:
done, we removed unnecessary parts from the example to avoid confusion